File: app/src/main/java/io/coderf/arklab/demo/repository/DemoPagingRepositoryImpl.java
package io.coderf.arklab.demo.repository;

import io.coderf.arklab.demo.api.ApiServiceHelper;
import io.coderf.arklab.demo.bean.NotificationMessageBean;

import java.util.Collections;
import java.util.List;

import io.reactivex.rxjava3.core.Observable;
import io.coderf.arklab.common.base.BaseView;
import io.coderf.arklab.common.inter.RetryService;
import io.coderf.arklab.common.repository.PagingRepositoryImpl;

/**
 * Created by fz on 2023/12/1 15:25
 * describe :
 */
public class DemoPagingRepositoryImpl extends PagingRepositoryImpl<ApiServiceHelper,NotificationMessageBean, BaseView> {

    public DemoPagingRepositoryImpl(RetryService retryService, BaseView baseView) {
        super(retryService, baseView);
    }

    public DemoPagingRepositoryImpl(BaseView baseView) {
        super(baseView);
    }

    @Override
    public Observable<List<NotificationMessageBean>> requestPaging(int currentPage, int pageSize) {
        NotificationMessageBean notificationMessageBean = new NotificationMessageBean();
        notificationMessageBean.setType("1");
        return sendRequest(
                apiService.getNewList(currentPage,pageSize,notificationMessageBean).map(response -> {
                    if(response.getList()==null){
                        return Collections.emptyList();
                    }
                    return response.getList();
                }),
                getApiRequestOptions());
    }
}

File: app/src/main/java/io/coderf/arklab/demo/repository/DemoPagingRepositoryImpl.kt
package io.coderf.arklab.demo.repository

import io.coderf.arklab.core.network.NetworkPagingRepository
import io.coderf.arklab.core.request.TokenRefresher
import io.coderf.arklab.demo.api.ApiServiceHelper
import io.coderf.arklab.demo.bean.NotificationMessageBean
import io.coderf.arklab.demo.bean.RequestNotificationBean

/**
 * Java 侧遗留的 Demo 分页仓库，已迁到新版 [NetworkPagingRepository]。
 * （原 Rx [PagingRepositoryImpl] 实现已移除。）
 *
 * @author fz
 * @version 1.0
 * @since 1.0
 * @updated 2026/9/1 22:51
 */
class DemoPagingRepositoryImpl(
    private val api: ApiServiceHelper,
    tokenRefresher: TokenRefresher? = null
) : NetworkPagingRepository<NotificationMessageBean, RequestNotificationBean>(
    tokenRefresher = tokenRefresher
) {

    override suspend fun fetchPage(page: Int, pageSize: Int, query: RequestNotificationBean): List<NotificationMessageBean> {
        val pageBean = api.getNewListSuspend(
            page,
            pageSize,
            query
        )
        return pageBean.list ?: emptyList()
    }
}
